c_src/filewatch_inotify.c

```diff
@@ -71,8 +69,23 @@ static ErlDrvData start(ErlDrvPort port, char *command)
     self->port = port;
     self->cooldown_ms = cooldown_ms;
 
+    // allocate memory for the name filed in each Message
+    for(i = 0; i < MAX_EVENTS; i++) {
```

Comment:
Note that C89 (might be C99) has allowed the declaration of variables in for loop headers (eg. `for (int i = 0; ...`) which will be scoped to the loop itself. In other words, I'd recommend scoping the `i` variable to each of the for loops in this function as opposed to reusing the variable.